### PR TITLE
ipvs: add connection timer operation APIs as below.

### PR DESCRIPTION
**Pull request by andrewhit**

### CODE REVIEW

File: include/ipvs/conn.h [hunk 1]
```diff
@@ -280,6 +281,36 @@ static inline void dp_vs_control_add(struct dp_vs_conn *conn, struct dp_vs_conn
...
+{
+    conn->flags |= DPVS_CONN_F_IN_TIMER;
+}
+
+static inline void
+dp_vs_conn_clear_in_timer(struct dp_vs_conn *conn)
+{
+    conn->flags &= ~DPVS_CONN_F_IN_TIMER;
+}
+
```
Comment (ywc689) [inline include/ipvs/conn.h]: What's the purpose to use the conn flag "DPVS_CONN_F_IN_TIMER"？ Is there any problem if the flag not used？

Comment (andrewhit) [reply include/ipvs/conn.h]: The purpose is for easy code logic and it is similar as the processing of DPVS_CONN_F_HASHED.

File: src/ipvs/ip_vs_conn.c [hunk 2]
```diff
@@ -495,10 +542,7 @@ static int conn_expire(void *priv)
     /* refcnt == 1 means we are the only referer.
      * no one is using the conn and it's timed out. */
     if (rte_atomic32_read(&conn->refcnt) == 1) {
-        if (conn->flags & DPVS_CONN_F_TEMPLATE)
-            dpvs_timer_cancel_nolock(&conn->timer, true);
-        else
-            dpvs_timer_cancel_nolock(&conn->timer, false);
+        dp_vs_conn_detach_timer(conn);
```
Comment (ywc689) [inline src/ipvs/ip_vs_conn.c]: "-nolock" timer api should be used in timer's callback.

File: src/ipvs/ip_vs_conn.c [hunk 3]
```diff
@@ -576,12 +620,10 @@ static int conn_expire(void *priv)
 
     /* some one is using it when expire,
      * try del it again later */
-    if (conn->flags & DPVS_CONN_F_TEMPLATE)
-        dpvs_timer_update_nolock(&conn->timer, &conn->timeout, true);
-    else
-        dpvs_timer_update_nolock(&conn->timer, &conn->timeout, false);
+    dp_vs_conn_refresh_timer(conn);
```
Comment (ywc689) [inline src/ipvs/ip_vs_conn.c]: "-nolock" timer api should be used in timer's callback.

File: src/ipvs/ip_vs_conn.c [hunk 4]
```diff
@@ -1038,10 +1073,7 @@ void dp_vs_conn_put(struct dp_vs_conn *conn)
 /* used in conn timer handler: conn_expire */
 static void dp_vs_conn_put_nolock(struct dp_vs_conn *conn)
 {
-    if (conn->flags & DPVS_CONN_F_TEMPLATE)
-        dpvs_timer_update_nolock(&conn->timer, &conn->timeout, true);
-    else
-        dpvs_timer_update_nolock(&conn->timer, &conn->timeout, false);
+    dp_vs_conn_refresh_timer(conn);
```
Comment (ywc689) [inline src/ipvs/ip_vs_conn.c]: "-nolock" timer api should be used here.

File: src/ipvs/ip_vs_conn.c (hunk 2 above)
Comment (andrewhit) [reply src/ipvs/ip_vs_conn.c]: Done.

File: src/ipvs/ip_vs_conn.c (hunk 4 above)
Comment (andrewhit) [reply src/ipvs/ip_vs_conn.c]: Done.

**Pull request merged**